synedrion/src/uint/signed.rs

```diff
@@ -202,6 +202,10 @@ impl<T: UintLike> Signed<T> {
         T::ModUint::conditional_select(&abs_mod, &-abs_mod, self.is_negative())
     }
 
+    pub fn secret_box(self) -> SecretBox<Signed<T>> {
```

Comment:
I am not certain about this method, I think its existence sort of provokes unsafe secret handling. Every time it is used we have a secret value being left on the stack.

Reply:
I thought that by passing `self` and moving ownership would remove that risk though. How can I
check? Do you think adding `#[inline(always)]` would help?